scapy/arch/unix.py

```diff
@@ -21,6 +21,20 @@
 ## Routes stuff ##
 ##################
 
+def guess_iface_name(netif):
+    """
+    We attempt to guess the name of interfaces that are truncated from the
+    output of ifconfig -l.
+    If there is only one possible candidate matching the interface name then we
+    return it. If there are none or more, then we return None.
+    """
+    f = os.popen('ifconfig -l')
+    ifaces = f.readlines()[0].strip().split(' ')
```

Comment:
Ok, I can change to use the with statement.

Regarding:

```
    for iface in ifaces:
        if iface.startswith(netif):
            return iface
 ```
I want to avoid returning an interface name when there are multiple matches as that means that I am actually not sure what interface is the correct one.